components/payments/internal/storage/bank_accounts.go

```diff
@@ -38,30 +38,48 @@ type bankAccount struct {
 	RelatedAccounts []*bankAccountRelatedAccount `bun:"rel:has-many,join:id=bank_account_id"`
 }
 
-func (s *store) BankAccountsUpsert(ctx context.Context, bankAccount models.BankAccount) error {
+func (s *store) BankAccountsUpsert(ctx context.Context, ba models.BankAccount) error {
 	tx, err := s.db.BeginTx(ctx, nil)
 	if err != nil {
 		return e("begin transaction", err)
 	}
 	defer tx.Rollback()
 
-	toInsert := fromBankAccountModels(bankAccount)
+	toInsert := fromBankAccountModels(ba)
 	// Insert or update the bank account
-	_, err = tx.NewInsert().
+	var idsUpdated []uuid.UUID
+	err = tx.NewInsert().
 		Model(&toInsert).
+		Column("id", "created_at", "name", "country", "metadata").
 		On("CONFLICT (id) DO NOTHING").
-		Exec(ctx)
+		Returning("id").
+		Scan(ctx, &idsUpdated)
```

Comment:
Couldn't you use `sql.Result.RowsAffected()` and `sql.Result.LastInsertId()` from the Exec result instead of changing the query to include a return value?